lnwallet/test/test_interface.go

```diff
@@ -3435,6 +3435,74 @@ func runTests(t *testing.T, walletDriver *lnwallet.WalletDriver,
 			// Bob.
 			aliceClient = chainConn.NewBitcoindClient()
 			bobClient = chainConn.NewBitcoindClient()
+
+		case "bitcoind-rpc-polling":
+			// Start a bitcoind instance.
+			tempBitcoindDir, err := ioutil.TempDir("", "bitcoind")
+			if err != nil {
+				t.Fatalf("unable to create temp directory: %v", err)
+			}
+			defer os.RemoveAll(tempBitcoindDir)
+			rpcPort := rand.Int()%(65536-1024) + 1024
```

Comment:
IIRC, we can instead use the "all interfaces" short cut here to obtain a port that we _know_ isn't taken. We may also want to propagate this elsewhere in this package, as I think some of our flakes are due to us trying to use a random port that's actually already taken.

Reply:
ah ok, asked around and seems you mean use `localhost:0`

Reply:
cant seem to get that to work... it causes the tests to panic.

could also try use something similar to NextAvailablePort but cant use that directly since it is in `lntest` and cause an import cycle. Since we seem to use `rpcPort := rand.Int()%(65536-1024) + 1024` in 2 different packages, maybe it makes sense to have a `NextAvailablePort` type function that is accessible by all other packages?

Reply:
What panic were you getting? Here's an example of us using this trick in the `brontide` tests: https://github.com/lightningnetwork/lnd/blob/master/brontide/noise_test.go#L32

If this doesn't work for w/e reason, then `NextAvailablePort` should be ok, but I'd like to understand _why_ as well.

Reply:
Ah I see what you mean. So then `NextAvailablePort` is likely the best approach, though it still leaves room for race conditions (multiple processes running and one of them happens to snag the port, etc).

Reply:
cool, the current diff does this 👍